### PR TITLE
Handle PortToConnect correctly, when same node recommendations are returned from ML API

### PR DESCRIPTION
**Pull request by reddyashish**
### Purpose

https://jira.autodesk.com/browse/DYN-5645

Now the correct port is connected when same node recommendations are returned from ML API. The results are displayed as list box and not as menu items. So we won't be able to provide sub menu functionality.

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB 

### Release Notes

Handle PortToConnect correctly, when same node recommendations are returned from ML API

### Reviewers

@QilongTang @zeusongit

### CODE REVIEW

File: src/DynamoCore/Search/SearchElements/NodeSearchElement.cs [hunk 1]
```diff
@@ -283,6 +283,14 @@ protected virtual IEnumerable<string> GenerateOutputParameters()
             inputParameters.Add(Tuple.Create(String.Empty, Properties.Resources.NoneString));
             return inputParameters;
         }
+
+        /// <summary>
+        ///    Clone the NodeSearchElement.
+        /// </summary>
+        public object Clone()
```
Comment (QilongTang) [inline src/DynamoCore/Search/SearchElements/NodeSearchElement.cs:290]: internal?

**Review by QilongTang — APPROVED** [review]

File: src/DynamoCoreWpf/ViewModels/Search/NodeAutoCompleteSearchViewModel.cs [hunk 2]
```diff
@@ -303,12 +303,18 @@ internal void DisplayMachineLearningResults()
...
                     if (result.Node.Type.NodeType.Equals(Function.FunctionNode))
                     {
+                        NodeSearchElement nodeSearchElement = null;
                         var element = zeroTouchSearchElements.FirstOrDefault(n => n.Descriptor.MangledName.Equals(result.Node.Type.Id));
 
-                        // Set PortToConnect for each element based on port-index and port-name
                         if (element != null)
                         {
-                            element.AutoCompletionNodeElementInfo = new AutoCompletionNodeElementInfo
+                            nodeSearchElement = (NodeSearchElement) element.Clone();
```
Comment (zeusongit) [inline src/DynamoCoreWpf/ViewModels/Search/NodeAutoCompleteSearchViewModel.cs:311]: So correct me if I am wrong here, we are now cloning the search element to have one instance for each port index that is returned in the ML results?

Comment (reddyashish) [reply src/DynamoCoreWpf/ViewModels/Search/NodeAutoCompleteSearchViewModel.cs:311]: We are cloning for each search suggestion as portToConnect is on each search element and it was being modified for same node suggestions.

File: src/DynamoCore/Search/SearchElements/NodeSearchElement.cs (hunk 1 above)
Comment (reddyashish) [reply src/DynamoCore/Search/SearchElements/NodeSearchElement.cs:290]: Has to be public as NodeSearchElement is extending ICloneable.

**Comment by reddyashish** [comment]
Created a new PR: https://github.com/DynamoDS/Dynamo/pull/13899

**Pull request closed without merging**